qpext/cmd/qpext/main.go

```diff
@@ -116,7 +202,6 @@ func scrape(url string, header http.Header, logger *zap.Logger) (io.ReadCloser,
 			logger.Error("Failed to parse timeout header", zap.Error(err), zap.String("timeout", timeoutString))
 		} else {
 			ctx, cancel = context.WithTimeout(ctx, timeout)
-			defer cancel()
```

Comment:
I think we still need this, failing to call the cancel func leaks the child resources.

Reply:
oh actually just realized that it is called outside of the `scrape` func

Reply:
yeah I think this was causing ctx cancel issues